Solnet.Serum/MarketManager.cs

```diff
@@ -0,0 +1,874 @@
+                OpenOrdersAccount = account;
+                action(account.Orders, slot);
+            }, _openOrdersAccount);
+        }
+
+        /// <inheritdoc cref="IMarketManager.SubscribeOpenOrders"/>
+        public void SubscribeOpenOrders(Action<IList<OpenOrder>, ulong> action) =>
+            SubscribeOpenOrdersAsync(action).Wait();
+
+        /// <inheritdoc cref="IMarketManager.UnsubscribeTradesAsync"/>
+        public async Task UnsubscribeTradesAsync()
+            => await _serumClient.UnsubscribeEventQueueAsync(_eventQueueSubscription.Address);
+
+        /// <inheritdoc cref="IMarketManager.UnsubscribeOrderBookAsync"/>
+        public async Task UnsubscribeOrderBookAsync()
+        {
+            await _serumClient.UnsubscribeOrderBookSideAsync(_bidSideSubscription.Address);
+            await _serumClient.UnsubscribeOrderBookSideAsync(_askSideSubscription.Address);
+        }
+
+        /// <inheritdoc cref="IMarketManager.UnsubscribeOpenOrdersAsync"/>
+        public async Task UnsubscribeOpenOrdersAsync()
+            => await _serumClient.UnsubscribeOpenOrdersAccountAsync(_openOrdersSubscription.Address);
+
+        #endregion
+
+        #region Order Management
+
+        /// <inheritdoc cref="IMarketManager.NewOrderAsync(Order)"/>
+        public async Task<SignatureConfirmation> NewOrderAsync(Order order)
+        {
+            if (_requestSignature == null)
+                throw new Exception("signature request method hasn't been set");
+
+            string blockHash = await GetBlockHash();
+
+            TransactionBuilder txBuilder = new TransactionBuilder()
+                .SetRecentBlockHash(blockHash)
+                .SetFeePayer(_ownerAccount);
+
+            PublicKey bAta = VerifyBaseAccountExists(txBuilder);
+            PublicKey qAta = VerifyQuoteAccountExists(txBuilder);
+            PublicKey ooa = await VerifyOpenOrdersExists(txBuilder);
+
+            order.ConvertOrderValues(_baseDecimals, _quoteDecimals, Market);
+
+            TransactionInstruction txInstruction = SerumProgram.NewOrderV3(
+                Market,
+                ooa,
+                order.Side == Side.Buy ? qAta : bAta,
+                _ownerAccount,
+                order,
+                _srmAccount);
+
+            TransactionInstruction settleIx = SerumProgram.SettleFunds(
+                Market,
+                ooa,
+                _ownerAccount,
+                bAta,
+                qAta);
+
+            byte[] txBytes = txBuilder
+                .AddInstruction(txInstruction)
+                .AddInstruction(settleIx)
+                .CompileMessage();
+
+            byte[] signatureBytes = _requestSignature(txBytes);
+
+            List<byte[]> signatures = new() { signatureBytes };
+            signatures.AddRange(_signers.Select(signer => signer.Sign(txBytes)));
+            _signers.Clear();
+
+            Transaction tx = Transaction.Populate(
+                Message.Deserialize(txBytes), signatures);

```

Comment:
can't we save the pre-signature transaction object and just add in the signature?